packages/material-ui/src/Select/Select.test.js

```diff
@@ -121,7 +121,7 @@ describe('<Select />', () => {
     });
 
     expect(handleBlur.callCount).to.equal(0);
-    expect(queryByRole('listbox', { hidden: false })).to.equal(null);
+    expect(queryByRole('listbox', { hidden: false })).not.to.equal(null);
```

Comment:
This is the unit test that failed after the code change and for which I'm not sure that this is the right way to handle it. I checked it manually and it seems to work as expected.

Reply:
I think this has to do with how react-transition-group works. It's likely hidden but still in the DOM at this point. After a single tick (`clock.tick(0)`) it should be removed. 

I'll take a though since that's still confusing to me. I was under the impression that change and close updates are batched but it seems like they aren't.

Reply:
The new behavior changed unrelated existing behavior: The test didn't test specifically for clicking the selected element but the behavior is still important: clicking the selected element should still close the Select. Since we had an early return in `handleItemClick` `update(false, event)` was never reached in this test.

The pushed fix hopefully makes it more obvious what the expected behavior is. I added an extra test specifically for this behavior.

Reply:
Ou ok, thanks for stepping in. I was also kind of afraid that this might alter some other existing behavior that wasn't covered by unit tests.